moveit_plugins/moveit_fake_controller_manager/src/moveit_fake_controller_manager.cpp

```diff
@@ -123,67 +119,77 @@ class MoveItFakeControllerManager : public moveit_controller_manager::MoveItCont
     }
   }
 
-  sensor_msgs::JointState loadInitialJointValues(XmlRpc::XmlRpcValue& param) const
+  sensor_msgs::JointState loadInitialJointValues() const
   {
     sensor_msgs::JointState js;
-
-    if (param.getType() != XmlRpc::XmlRpcValue::TypeArray || param.size() == 0)
-    {
-      ROS_ERROR_ONCE_NAMED("loadInitialJointValues", "Parameter 'initial' should be an array of (group, pose) "
-                                                     "structs.");
-      return js;
-    }
-
+    XmlRpc::XmlRpcValue initial;
     robot_model_loader::RobotModelLoader robot_model_loader(ROBOT_DESCRIPTION);
     const robot_model::RobotModelPtr& robot_model = robot_model_loader.getModel();
     typedef std::map<std::string, double> JointPoseMap;
     JointPoseMap joints;
 
-    for (int i = 0, end = param.size(); i != end; ++i)
+    robot_model->getVariableDefaultPositions(joints);
```

Comment:
I think I agree with @rhaschke that we should not publish the full joint state of the robot model. I would only publish values for joints that the fake controller is actually managing.

Reply:
So we should move `loadInitialJointValues` after creating the controllers, right .?